docs/synapse/userguides/storm_ref_automation.rstorm

```diff
@@ -651,72 +901,225 @@ Storm each time. The Storm can be as simple or advanced as you like.
+  take some time to run, which can cause the Synapse interface (the Storm CLI or the Optic UI) to "block"
+  for the user until the Storm executes in full. The macro code can be run asynchronously using the Storm
+  :ref:`storm-background` command. This "unblocks" the CLI / UI, but the full set of results from enrichment
+  will not be available until the macro finishes executing.
+
+
+**Modify macro permissions - examples**
 
+You can modify permissions on a macro using the :ref:`stormlibs-lib-macro-grant` library to grant (or revoke)
+access. The :ref:`stormlibs-lib-auth-users-byname` and :ref:`stormlibs-lib-auth-roles-byname` libraries can be
+used to obtain a user or role object and retrieve the associated identifier (iden).
 
-- Add a macro named ``check.c2`` that takes an inbound set of ``file:bytes`` nodes and returns any FQDNs
-  that the files query and any IPv4 addresses the files connect to. Use a filter in the macro to ensure that
-  the macro code only attempts to process inbound ``file:bytes`` nodes.
+.. storm-pre:: auth.user.add 'ron the cat' --email ron@vertex.link
+.. storm-pre:: auth.user.add 'maeve' --email maeve@vertex.link
+.. storm-pre:: auth.role.add interns
+.. storm-pre:: auth.role.add 'cattribution analysts'
+.. storm-pre:: auth.user.grant 'ron the cat' 'cattribution analysts'
+.. storm-pre:: auth.user.grant maeve interns
 
-.. storm-cli:: macro.set check.c2 ${ +file:bytes | tee { -> inet:dns:request :query:name -> inet:fqdn | uniq } { -> inet:flow:src:exe :dst:ipv4 -> inet:ipv4 | uniq } }
+`Make user "ron the cat" co-admin of the "sinkhole.check" macro:`
 
-- Add a macro named ``enrich`` that takes any node as input and uses a ``switch`` statement to call Storm
-  commands for third-party services able to enrich a given form (line breaks and indentations used for readability).
-  (**Note:** Synapse does not include third-party services / connectors in its open source distribution;
-  the macro assumes such services have been implemented.)
+.. storm-cli:: $user=$lib.auth.users.byname('ron the cat') $iden=$user.iden $lib.macro.grant(sinkhole.check,users,$iden,3)
 
-.. storm-pre:: $pkg=$lib.dict(name='docs', version='0.0.1', commands=($lib.dict(name=whois, storm=${} ),)) $lib.print($pkg) $lib.pkg.add($pkg)
-.. storm-pre:: $pkg=$lib.dict(name='docs', version='0.0.1', commands=($lib.dict(name=malware, storm=${} ),)) $lib.print($pkg) $lib.pkg.add($pkg)
-.. storm-pre:: $pkg=$lib.dict(name='docs', version='0.0.1', commands=($lib.dict(name=revwhois, storm=${} ),)) $lib.print($pkg) $lib.pkg.add($pkg)
-.. storm-multiline:: ENRICH="macro.set enrich ${ switch $node.form() {\n\n    /* You can put comments in macros!!! */\n\n    \"inet:fqdn\": { | whois | pdns | malware }\n    \"inet:ipv4\": { | pdns }\n    \"inet:email\": { | revwhois }\n    *: { }\n} }"
-.. storm-cli:: MULTILINE=ENRICH
+`Deny the "interns" role access to the "enrich" macro:`
```

Comment:
For completeness can we add a role + edit permission example?

Reply:
added